server/src/main/java/org/gluu/oxtrust/ldap/service/ApplicationFactory.java

```diff
@@ -96,4 +102,16 @@ public PersistenceEntryManagerFactory getPersistenceEntryManagerFactory() {
         return persistenceEntryManagerFactory;
     }
 
+    @Produces @ApplicationScoped
+    @Named("facesMessageSourceProvider")
+    public MessageSourceProvider facesMessageSourceProvider() {
+        return new FacesMessageSourceProvider(facesMessages);
+    }
+
+    @Produces @ApplicationScoped
+    @Named("smtpStatusFactory")
+    public SmtpStatusFactory smtpStatusFactory() {
+        return new SmtpStatusFactory(mailService, facesMessageSourceProvider(), smtpConfigurationService);
```

Comment:
Why we pass references to CDI beans to SmtpStatusFactory? Why we can't allow inject them in SmtpStatusFactory?

Is there any reason of using `@Produces` here?

Reply:
Yes,
In fact i'm using a **faces** implementation to get properties in the api which is not supposed to be faces aware.
If I use the Inject annotation, I'll have to use a naming "facesMessageSourceProvider", hence they will be coupled

Reply:
In my code I'm trying to avoid passing CDI managed bean into non CDI bean. In reality you pass proxy class which Weld can destroy if needed. In your case ApplicationScoped will gurantee that the reference will be valid. But... why not define:
```
@Named("facesMessageSourceProvider")
@ApplicationScoped
public class MessageSourceProvider {

@Inject
private FacesMassages facesMassages;
...